NotebookAnalyzers/NotebooksAnalysisResults.ipynb

```diff
@@ -0,0 +1,247 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Notebooks Analysis Results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Vengono importate\n",
+    "##### \"os\" per la gestione dei file,\n",
+    "##### \"pandas\" per la creazione del dataframe con i risultati di analisi\n",
+    "##### \"utils\", il modulo contenente le operazionalizzazioni delle best practice"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import utils\n",
+    "import pandas"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Ogni target notebook viene analizzato applicando le funzioni per l'analisi della qualità che rispecchiano le best practice individuate"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data=[]\n",
+    "for filename in os.listdir(\"../TargetNotebooks/\"):\n",
+    "       if filename.endswith(\".ipynb\"):\n",
+    "              py_code = utils.notebook_to_code(filename)\n",
+    "              notebook_dict = utils.notebook_to_json(filename)\n",
+    "              distribution = utils.markdown_distribution(notebook_dict)\n",
+    "              if utils.markdown_distribution(notebook_dict) is not None:\n",
```

Comment:
Qui richiami per due volte la stessa funzione.

A parte questo, è davvero necessario diramare in questo modo l'esecuzione dello script? La funzione `utils.markdown_distribution` non restituisce un array di oggetti `None` nel caso in cui sia impossibile calcolare le percentuali (e.g., `[None, None, None, None]`? Se così fosse, non ci sarebbe più bisogno dell'`if` statement: il caso particolare verrebbe gestito dalla funzione che ne ha la responsabilità.